test/cli_test.js

```diff
@@ -15,19 +15,16 @@ const cliRun = require('../lib/cli').run;
 
 suite('cli', () => {
 
-  test('unknown cmd parameter should not throw exception', (done) => {
-    var argv = process.argv;
-
+  test('unknown cmd parameter should not throw exception', () => {
+    const originalArgv = process.argv;
     process.argv = ["node", "polyserve", "--unknown-parameter"];
-    var result = cliRun();
 
-    result.then(function() {
-      process.argv = argv;
-      done();
-    }, function(err) {
-      process.argv = argv;
-      done(err);
-    });
+    try {
+      return cliRun();
```

Comment:
I think what you want here is:

`return await cliRun();`

With just returning `cliRun` directly you're resetting the argv before the first microtask, which probably works for now but is fragile